packages/core/src/shared/crashMonitoring.ts

```diff
@@ -154,15 +173,19 @@ export class CrashMonitoring {
  */
 class Heartbeat {
```

Comment:
Hearbeat is a concept that continues to come up, e.g. we have another heartbeat for codecatalyst: https://github.com/aws/aws-toolkit-vscode/blob/5573cbb1f56f9ec38a1c20572113eda1e5556d90/packages/core/src/codecatalyst/devEnv.ts#L65

Even if the codecatalyst heartbeat is too different to be useful here, is this new Heartbeat not amenable to generalization? It is a general concept that is needed by different features that want to emit data on a timed basis. 

Do we have a backlog item to pull this out / generalize it? Are features easily able to attach data to a `toolkit_heartbeat` metric, or did we name the metric `crashreporting_x` (not intuitive)?

Reply:
There isn't much reuse between Codecatalyst heartbeat and this heartbeat atm. This one heartbeats on an interval, and the other heartbeats when the user performs actions in the IDE.

We could make an interface for the common apis, but this doesn't deduplicate anything existing code